### PR TITLE
Use GitHub actions

### PR DESCRIPTION
**Pull request by copelco**
* Remove TravisCI
* Add GitHub actions
* Repo cleanup and re-run `black`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,4 @@
-pdfrw==0.4
-jupyterlab==0.35.4
```
Comment (robert-w-gries) [inline requirements.txt:2]: I think @georgehelman uses`jupyterlab` in his workflow. Maybe we could make a separate requirements file for local development stuff

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Cache pip
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements/**.txt') }}
```
Comment (robert-w-gries) [inline .github/workflows/test.yml:21]: Unless we make changes to the requirements file structure, we need to change this line:

```diff
- key: ${{ runner.os }}-pip-${{ hashFiles('requirements/**.txt') }}
+ key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -3,4 +3,10 @@ repos:
...
     hooks:
     - id: black
-      language_version: python3
+      language_version: python3.8
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v3.4.0
+  hooks:
+    - id: check-merge-conflict
+    - id: debug-statements
+    - id: detect-private-key
```
Comment (robert-w-gries) [inline .pre-commit-config.yaml:12]: These looks nice. Is this something we should add to the `dear-petition` repo later?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+[tool.black]
+line-length = 120
+target-version = ['py38']
+include = '\.pyi?$'
+exclude = '''
```
Comment (robert-w-gries) [inline pyproject.toml:5]: Nitpick: It looks like `black` [automatically ignores files in the gitignore](https://github.com/psf/black/pull/878) so you might be able to clean this up a bit.

**Review by robert-w-gries — CHANGES_REQUESTED** [review]
Two required changes in the cache action `key` and not disrupting George's dev workflow, otherwise the rest of the comments are open question

File: requirements.txt (hunk 1 above)
Comment (copelco) [reply requirements.txt:2]: Yeah, I use it too, but I don't think it needs to be apart of our CI pipeline. It's easy enough to pip install manually if needed.

File: .github/workflows/test.yml (hunk 2 above)
Comment (copelco) [reply .github/workflows/test.yml:21]: Good catch! Thanks

**Review by robert-w-gries — APPROVED** [review]
Merge at will

**Pull request merged**